src/V3Partition.cpp

```diff
@@ -2010,13 +2010,27 @@ class PartPackMTasks;
 // (attributes).
 class ThreadSchedule final {
 public:
+    // CONSTANTS
+    static constexpr uint32_t UNASSIGNED = 0xffffffff;
+
+    // TYPES
+    struct MTaskState {
+        uint32_t completionTime = 0;  // Estimated time this mtask will complete
+        uint32_t threadId = UNASSIGNED;  // Thread id this MTask is assigned to
+        const ExecMTask* nextp = nullptr;  // Next MTask on same thread after this
+    };
+
+    // MEMBERS
     // Allocation of sequence of MTasks to threads. Can be considered a map from thread ID to
     // the sequence of MTasks to be executed by that thread.
     std::vector<std::vector<const ExecMTask*>> threads;
 
     // Map from MTask to ID of thread it is assigned to.
     std::unordered_map<const ExecMTask*, uint32_t> threadId;
 
+    // State for each mtask.
+    std::unordered_map<const ExecMTask*, MTaskState> mtaskState;
```

Comment:
Actually, that mistake was intentional as these are "properties", and I was lazy to write no-op accessors :blush:

Reply:
I'm fine if we keep this as is then.

Reply:
Also note that once a ThreadSchedule object is constructed by the packing algorithm it can only exist as a `const` object, and only the packing algo can construct one (due to the `friend` declaration and private constructors). The idea was that the packing algo will build one of these poking around in the internals freely, but once you got one of these from the packing algo, it is read only, hence the "property" syntax seemed cleaner, but having just explained that that is possibly too much cognitive load on the developer.

Reply:
I'll keep them without the `m_` prefix for now.